### PR TITLE
Replace deprecated OpenAI models

### PR DESCRIPTION
**Pull request by bilgeyucel**

### CODE REVIEW

File: integrations/basic-agent-memory.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ memory_tool = Tool(name="Memory",
                    pipeline_or_node=memory_node,
                    description="Your memory. Always access this tool first to remember what you have learned.")
 
-prompt_node = PromptNode(model_name_or_path="text-davinci-003",
```
Comment (julian-risch) [inline integrations/basic-agent-memory.md:54]: recommended replacement is `gpt-3.5-turbo-instruct` https://platform.openai.com/docs/deprecations/instructgpt-models

**Review by julian-risch — CHANGES_REQUESTED** [review]
We should replace text-davinci-003 with gpt-3.5-turbo-instruct
otherwise the PR looks good to me.

File: integrations/basic-agent-memory.md (hunk 1 above)
Comment (TuanaCelik) [reply integrations/basic-agent-memory.md:54]: do you think this should be the replacement for all text-davinci-003 models or just in this integration? And if yes, why?

Comment (julian-risch) [reply integrations/basic-agent-memory.md:54]: For all of them. It's what OpenAI recommends. They are also saying that `gpt-3.5-turbo-instruct` was trained in a similar way to `text-davinci-003`. I would assume that `gpt-3.5-turbo-instruct` is better at following instructions than `gpt-3.5-turbo`.

**Pull request merged**